apollo-router-core/src/query_planner/mod.rs

```diff
@@ -361,7 +360,8 @@ mod fetch {
                 context: Object::default(),
             };
 
-            let response = service
+            let mut locked = service.lock().await;
```

Comment:
I know it s ugly but it compiles let s figure out a better way i guess 😭

Reply:
that will not fit with tower's model: a Service is not meant to be shared. If we want to have multiple instances of the same kind of service, we need a ServiceBuilder or, as I did previously, a registry that holds cloneable instances of Service (or can create new ones on the fly) and returns an owned Service instance. Calling [oneshot](https://docs.rs/tower/latest/tower/trait.ServiceExt.html#method.oneshot) solves other issues with the compiler complaining that the future returned by `call` borrows the Service. Oneshot consumes the service, calls `ready` then `call` and it's well packaged.

And now you know why I was busy trying to assemble types and lifetimes yesterday :grin:

Reply:
Haha true, for now I was just trying to get the branch back to a state where it compiles, let’s write an elegant way tomorrow 😁

Reply:
Oneshot looks amazing thb

Reply:
Ok thanks I will check from there